### PR TITLE
ACSP-173 developer tools

### PR DESCRIPTION
**Pull request by jhadobe**
This PR adds development info to the Starter Kit docs.

### CODE REVIEW

File: src/pages/app-development/starter-kit/actions-overview.md [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+* `event ingestion`
+* `synchronous webhook`
+
+The `consumer` and `event handler` actions implement the business logic needed to synchronize data between Commerce and the external systems being integrated.
+
+Additionally, boilerplate code, responses, and samples for event ingestion and synchronous webhook actions are provided.
+
+<InlineAlert variant="info" slots="text"/>
+
+The responses for each action are configured for the [Commerce webhook module](https://developer.adobe.com/commerce/extensibility/webhooks/). To customize the responses for other implementations, modify the `response.js` file.
```
Comment (rojoangel) [inline src/pages/app-development/starter-kit/actions-overview.md:28]: Typo: the file name is `responses.js`

Comment (rojoangel) [inline src/pages/app-development/starter-kit/actions-overview.md:28]: I don't think this statement is accurate for all actions. 
Only the `synchronous webhook` responses are configured to work with the Commerce webhook module.
Responses for the other actions are configured to work nicely with thethe AIO Events registration built-in retry mechanism.

File: src/pages/app-development/starter-kit/actions-overview.md [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+
+## `consumer` action
+
+The main purpose of this action is to route events to the [`event handler` action](#event-handler-action). Normally, this routing is determined by the name of the event received.
+
+The `consumer` action is subscribed to a set of events. In many cases, all the events originate from the same entity, such as the `product` entity. However, there are examples where a consumer receives events from multiple entities belonging to the same "domain", such as `order` and `shipment`. When the event provider receives an event, this runtime action will be automatically activated.
+
+The response returned by a `consumer` action is expected to be consistent with the response received from the activation of the subsequent `event handler` action. For example, if the `event handler` action returns an `HTTP/400` status, the consumer action is expected to respond with the same status.
```
Comment (rojoangel) [inline src/pages/app-development/starter-kit/actions-overview.md:36]: If you want to add additional details about this point. That consistency is built in to ensure proper event processing retry based on the response received from the `event handler` action.

File: src/pages/app-development/starter-kit/actions-overview.md [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+
+### Failure
+
+```js
+// ./actions/responses.js#webhookSuccessResponse
+return {
+  error: {
+    statusCode: 200,
+    body : {
+      op: 'exception'
```
Comment (rojoangel) [inline src/pages/app-development/starter-kit/actions-overview.md:193]: For completeness, the response can include a `message` too
```suggestion
      op: 'exception'
      message: 'YOUR ERROR MESSAGE'
```

File: src/pages/app-development/starter-kit/developer-tools.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+The `actionparams` object allows you to pass values from your environment variables to an action. Add the parameter to your `.env` file:
+
+`YOUR_PARAM=your_value`
+
+To pass the parameters to the action, add the parameter as an `input` under the corresponding action in the `actions/{entity}/../actions.config.yaml` file:
+
+```yaml
+{action name}:
+  function: commerce/{action name}/index.js
```
Comment (rojoangel) [inline src/pages/app-development/starter-kit/developer-tools.md:28]: I think it should be
```suggestion
  function: {action name}/index.js
```

File: src/pages/app-development/starter-kit/developer-tools.md [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+  web: 'no'
+  runtime: nodejs:16
+  inputs:
+    LOG_LEVEL: debug
+    YOUR_PARAM: $YOUR_PARAM_ENV
```
Comment (rojoangel) [inline src/pages/app-development/starter-kit/developer-tools.md:33]: Previously the param in the `.env` file has been defined as `YOUR_PARAM` in line 22. For consistency, either:
- change this line to 
```YOUR_PARAM: $YOUR_PARAM```, or
- change line 22 to
```YOUR_PARAM_ENV=your_value```

File: src/pages/app-development/starter-kit/index.md [hunk 6]
```diff
@@ -9,3 +9,79 @@ keywords:
...
+- An `actions.config.yaml` file. This file declares the runtime actions responsible for handling the events for an entity originating in Commerce or the external system.
+
+Individual directories that define actions contain the following files:
+
+File | Purpose
+--- | ---
+`index.js` | Contains the main method that gets invoked when handling an event. It is responsible for coordinating the different activities involved in that handling, such as validating the incoming payload, transforming the payload to the target API, and interacting with the target API.
+`pre.js` and `post.js` | Provide convenient extension points to introduce custom business logic before and after interacting with the target API.
+`sender.js` | Implements the logic to interact with the target API so that the changes are propagated. For actions in the `external` directory, Commerce APIs are the target. Actions in the `commerce` directory target external APIs.
+`schema.json` | (`external` directories only) Defines the schema to be validated.
```
Comment (rojoangel) [inline src/pages/app-development/starter-kit/index.md:75]: `Defines the schema to validate the incoming event payload.`, maybe?

**Review by rojoangel — APPROVED** [review]
Nice one! ❤️

**Review by keharper — APPROVED** [review]

**Pull request merged**